### PR TITLE
ref(insights): Wrap mobile insights module in `ModulePageProviders`

### PR DESCRIPTION
**Pull request by gggritso**
Preparation for URL and breadcrumb changes. The Mobile modules don't wrap pages in `ModulePageProviders`. I need the wrapper so that I can wholesale change the base URL for all modules at once without doing awkward search-and-replace.

Best enjoyed with whitespace off.

## Changes

- Fix typing in `ModulePageProviders`
    
    Delegate to `Feature` for anything passed through.

- Wrap mobile modules in `ModulePageProviders`
    
    `ModulePageProviders` includes `Feature`, `SentryDocumentTitle`, and
    `PageFiltersContainer`, so it's not necessary to wrap in those
    components manually.

### CODE REVIEW

**Comment by gggritso** [comment]
@narsaynorath could you take a look at this? Would love a second set of eyes to make sure the feature flag names and page titles are still correct

File: static/app/views/performance/mobile/ui/screenSummary/index.tsx [hunk 1]
```diff
@@ -72,72 +71,82 @@ function ScreenSummary() {
...
-export default ScreenSummary;
+function PageWithProviders() {
+  const location = useLocation<Query>();
+
+  const {transaction} = location.query;
+
+  return (
+    <ModulePageProviders
+      title={[transaction, t('Screen Loads')].join(' — ')}
+      baseURL="/performance/browser/app-startup"
```
Comment (narsaynorath) [inline static/app/views/performance/mobile/ui/screenSummary/index.tsx:141]:
```suggestion
      baseURL="/performance/mobile/ui/spans"
```

File: static/app/views/performance/mobile/appStarts/screenSummary/index.tsx [hunk 2]
```diff
@@ -109,120 +109,130 @@ function ScreenSummary() {
...
+}
+
+function PageWithProviders() {
+  const location = useLocation<Query>();
+
+  const {transaction} = location.query;
+
+  return (
+    <ModulePageProviders
+      title={[transaction, ROUTE_NAMES['app-startup']].join(' — ')}
```
Comment (narsaynorath) [inline static/app/views/performance/mobile/appStarts/screenSummary/index.tsx:226]: It looks like this `ROUTE_NAMES` constant is only really used by mobile modules. If you wanted to remove the constant and just define the title inline (i.e. `t(App Starts)`) then that removes another file from the old starfish path

File: static/app/views/performance/mobile/appStarts/screenSummary/index.tsx [hunk 3]
```diff
@@ -109,120 +109,130 @@ function ScreenSummary() {
...
+      baseURL="/performance/browser/app-startup"
```
Comment (narsaynorath) [inline static/app/views/performance/mobile/appStarts/screenSummary/index.tsx:227]:
```suggestion
      baseURL="/performance/mobile/app-startup/spans"
```

I think this is the appropriate path for this file. This component renders the drill down into a screen which uses the `/performance/mobile/app-startup/spans` URL. Unless this component uses `baseURL` as the module list page then the URL should end with `/spans`

File: static/app/views/performance/mobile/appStarts/index.tsx [hunk 4]
```diff
@@ -1,25 +1,34 @@
...
-          docsUrl:
-            'https://docs.sentry.io/product/performance/mobile-vitals/app-starts/#minimum-sdk-requirements',
-        }}
-        content={<AppStartup chartHeight={200} />}
-        title={ROUTE_NAMES['app-startup']}
-      />
-    </Feature>
+    <ModulePageProviders
+      title={ROUTE_NAMES['app-startup']}
+      baseURL="/performance/browser/app-startup"
```
Comment (narsaynorath) [inline static/app/views/performance/mobile/appStarts/index.tsx:26]:
```suggestion
      baseURL="/performance/mobile/app-startup"
```

**Review by narsaynorath — COMMENTED** [review]
Just a few spots where the URL is still referencing `browser` instead of `mobile`. Also gave a suggestion to remove another old starfish file. Feel free to re-request a review and I can do a quick click through of the modules to make sure everything's as expected. Thanks @gggritso!

File: static/app/views/performance/mobile/appStarts/screenSummary/index.tsx (hunk 2 above)
Comment (gggritso) [reply static/app/views/performance/mobile/appStarts/screenSummary/index.tsx:226]: Ah good call, I can take a look separately

**Comment by gggritso** [comment]
@narsaynorath thanks, good catches! Copy-pasting too fast on my side. The `baseURL` isn't in use yet, so I'll probably have to tweak those a bit once I actually use them for routing, too 👍🏻

**Review by narsaynorath — APPROVED** [review]

**Pull request merged**